File: output/gpt_4o/avatar/direct_translation/python/java/successful_translation/codeforces_678_B.java

import java.util.Scanner;

public class codeforces_678_B {
    public static String leapyear(int n) {
        if (n % 400 == 0 || (n % 4 == 0 && n % 100 != 0)) {
            return "y";
        } else {
            return "n";
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        String d = "f";
        String ans;
        if (n % 400 == 0 || (n % 4 == 0 && n % 100 != 0)) {
            ans = "ly";
        } else {
            ans = "nly";
        }
        while (true) {
            if (n % 400 == 0 || (n % 4 == 0 && n % 100 != 0)) {
                if (d.equals("m")) {
                    d = "w";
                    n += 1;
                } else if (d.equals("t")) {
                    d = "th";
                    n += 1;
                } else if (d.equals("w")) {
                    d = "f";
                    n += 1;
                } else if (d.equals("th")) {
                    d = "sa";
                    n += 1;
                } else if (d.equals("f")) {
                    d = "s";
                    n += 1;
                } else if (d.equals("sa")) {
                    d = "m";
                    n += 1;
                } else if (d.equals("s")) {
                    d = "t";
                    n += 1;
                }
            } else {
                if (d.equals("m")) {
                    d = "t";
                    n += 1;
                } else if (d.equals("t")) {
                    d = "w";
                    n += 1;
                } else if (d.equals("w")) {
                    d = "th";
                    n += 1;
                } else if (d.equals("th")) {
                    d = "f";
                    n += 1;
                } else if (d.equals("f")) {
                    d = "sa";
                    n += 1;
                } else if (d.equals("sa")) {
                    d = "s";
                    n += 1;
                } else if (d.equals("s")) {
                    d = "m";
                    n += 1;
                }
            }
            if (d.equals("f") && ans.equals("ly") && leapyear(n).equals("y")) {
                break;
            } else if (d.equals("f") && ans.equals("nly") && leapyear(n).equals("n")) {
                break;
            }
        }
        System.out.println(n);
    }
}

File: dataset/avatar/Python/Code/codeforces_678_B.py
def leapyear ( n ) :
    if ( n % 400 == 0 or ( n % 4 == 0 and n % 100 != 0 ) ) :
        return "y"
    else :
        return "n"
n = int ( input ( ) )
d = "f"
if ( n % 400 == 0 or ( n % 4 == 0 and n % 100 != 0 ) ) :
    ans = "ly"
else :
    ans = "nly"
while ( True ) :
    if ( n % 400 == 0 or ( n % 4 == 0 and n % 100 != 0 ) ) :
        if ( d == "m" ) :
            d = "w"
            n += 1
        elif ( d == "t" ) :
            d = "th"
            n += 1
        elif ( d == "w" ) :
            d = "f"
            n += 1
        elif ( d == "th" ) :
            d = "sa"
            n += 1
        elif ( d == "f" ) :
            d = "s"
            n += 1
        elif ( d == "sa" ) :
            d = "m"
            n += 1
        elif ( d == "s" ) :
            d = "t"
            n += 1
    else :
        if ( d == "m" ) :
            d = "t"
            n += 1
        elif ( d == "t" ) :
            d = "w"
            n += 1
        elif ( d == "w" ) :
            d = "th"
            n += 1
        elif ( d == "th" ) :
            d = "f"
            n += 1
        elif ( d == "f" ) :
            d = "sa"
            n += 1
        elif ( d == "sa" ) :
            d = "s"
            n += 1
        elif ( d == "s" ) :
            d = "m"
            n += 1
    if ( d == "f" and ans == "ly" and leapyear ( n ) == "y" ) :
        break
    elif ( d == "f" and ans == "nly" and leapyear ( n ) == "n" ) :
        break
print ( n )

